python/cudf/cudf/tests/test_index.py

```diff
@@ -274,21 +275,92 @@ def test_set_index_as_property():
     assert_eq(head.index, idx[:5])
 
 
+@pytest.mark.parametrize("deep", [True, False])
```

Comment:
Your tests aren't really testing whether the index is deep copied or not. Would suggest removing it from the parameterization and having a separate test for deep copying or not.

Reply:
Sure. Should I print out their underlying `_values` object address and assert they are not the same location for `deep=True`, and the same location otherwise?

Reply:
I think we should leave these tests as is and remove the deep parameter from them, then in a new test check specifically that we're deep copying by checking equality between the underlying column objects in `._values`